File: src/contests/cg/puzzle/python3/the-river-i-.py
# 🎮 CodinGame Puzzle - the-river-i-
# https://www.codingame.com/training/easy/the-river-i-


def digit_sum(n):
    s = 0
    while n > 0:
        s += n % 10
        n //= 10
    return s


a = int(input())
b = int(input())
while a != b:
    if a < b:
        a += digit_sum(a)
    else:
        b += digit_sum(b)
print(a)

File: src/contests/cg/puzzle/haskell/the-river-i-.hs
-- 🎮 CodinGame Puzzle - the-river-i-
-- https://www.codingame.com/training/easy/the-river-i-

main :: IO ()
main = do
  a <- readLn :: IO Integer
  b <- readLn :: IO Integer
  let digitSum x = sum (map (\c -> fromIntegral (fromEnum c - fromEnum '0')) (show x)) :: Integer
      go x y
        | x == y = x
        | x < y = go (x + digitSum x) y
        | otherwise = go x (y + digitSum y)
  print (go a b)
